RHIDP-5139: Installing RHDH container image plugins (#831)
redhat-developer/red-hat-developers-documentation-rhdh: modules/dynamic-plugins/proc-install-plugins-rhdh-container-image.adoc

[id="proc-install-plugins-rhdh-container-image_{context}"]
= Installing plugins included in {product-very-short} container image

> In the {product-very-short} container image, a set of dynamic plugins is pre-loaded to enhance functionality. However, due to mandatory configuration requirements, most of the plugins are disabled.

Reviewer: We have been replacing _pre-loaded_ with _preinstalled_ to comply with the IBM style guide.
"Do not use preloaded to refer to software that is already installed; use preinstalled."

> In the {product-very-short} container image, a set of dynamic plugins is pre-loaded to enhance functionality. However, due to mandatory configuration requirements, most of the plugins are disabled.

Reviewer: We have been replacing pre-loaded with preinstalled to comply with the IBM style guide.
"Do not use preloaded to refer to software that is already installed; use preinstalled."
```suggestion
In the {product-very-short} container image, a set of dynamic plugins is preinstalled to enhance functionality. However, due to mandatory configuration requirements, most of the plugins are disabled.
```

Reviewer 2: @hmanwani-rh Yes, we should stick with "preinstalled".

Author: I don't think it makes sense to describe this section as "installing a preinstalled plugin". I'll do some more research on the topic and come up with something more relevant.

> You can activate and configure the plugins in the {product-very-short} container image, including how to manage the default configuration, set necessary environment variables, and ensure the proper functionality of the plugins within your application.

Reviewer 2: @hmanwani-rh Generally we refer to "enabling" and "disabling" preinstalled plugins. There a few similar mentions of "activating" plugins in this PR. I'd suggest sticking with enable/disable.

```suggestion
You can enable and configure the plugins in the {product-very-short} container image, including how to manage the default configuration, set necessary environment variables, and ensure the proper functionality of the plugins within your application.
```

.Prerequisites

> * You have access to the link:https://github.com/janus-idp/backstage-showcase/blob/main/dynamic-plugins.default.yaml[`dynamic-plugins.default.yaml`] file, which lists all preloaded plugins.

Reviewer: ```suggestion
* You have access to the link:https://github.com/janus-idp/backstage-showcase/blob/main/dynamic-plugins.default.yaml[`dynamic-plugins.default.yaml`] file, which lists all preinstalled plugins.
```

Author: I don't think it makes sense to describe this section as "installing a preinstalled plugin". I'll do some more research on the topic and come up with something more relevant.

* You have deployed the {product-very-short} application, and have access to the logs of the `install-dynamic-plugins` init container.
* You have the necessary permissions to modify plugin configurations and access the application environment.
* You have identified and set the required environment variables referenced by the plugin's default configuration. These environment variables must be defined in the Helm Chart or Operator configuration.

.Procedure
. Locate the link:https://github.com/janus-idp/backstage-showcase/blob/main/dynamic-plugins.default.yaml[`dynamic-plugins.default.yaml`] file in your setup.
. Start your {product-very-short} application and access the logs of the `install-dynamic-plugins` init container within the {product-very-short} pod.

> . Look for log entries indicating plugins that are disabled by default. For example:

Reviewer 3: I don't think that getting list of the plugins from logs is a user-friendly way to get the list of available plugins. 

We list of the plugins in our documentation (https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.3/html/dynamic_plugins_reference/con-preinstalled-dynamic-plugins) so we should probably guide users use this instead parsing logs.

+
--
.Example log entry for a disabled plugin
[source,sql]
----
======= Skipping disabled dynamic plugin ./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-github-dynamic
----
--
. Open the plugin configuration file and locate the plugin entry you want to activate.
. Modify the `disabled` field to `false` and specify the package name as follows:
+
--
.Example plugin configuration
[source,yaml]
----
plugins:
  - disabled: false
    package: ./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-github-dynamic
----
--

> . Optional: Replace the default plugin configuration from the `dynamic-plugins.default.yaml` file by adding a `pluginConfig` entry to the plugin configuration file.

Reviewer 3: Where can users find dynamic-plugins.default.yaml file?
